examples/elf/Makefile

```diff
@@ -51,18 +51,11 @@ ROOTDEPPATH := --dep-path tests
 
 VPATH += :tests
 
-ifeq ($(CONFIG_EXAMPLES_ELF_ROMFS),y)
-  FSIMG_HDR = $(TESTS_DIR)/romfs.h
-else
-  FSIMG_HDR = $(TESTS_DIR)/cromfs.h
-endif
-
-elf_main.c: tests/symtab.c
-
-tests/symtab.c tests/dirlist.h $(FSIMG_HDR):
```

Comment:
@masayuki2009 why we change the target to context here? What problem can be fixed by this change?

Reply:
@xiaoxiang781216 This change was needed to fix such as

riscv64-unknown-elf-ld: pthread.o: file not recognized: file truncated                                                                                        
Makefile:82: recipe for target 'pthread' failed                                                                                                               
make[4]: *** [pthread] Error 1                                                                                                                                

Actually there was a similar error in 
https://github.com/apache/incubator-nuttx-apps/pull/118#issuecomment-598017857

Reply:
I think this error is because:
tests/symtab.c tests/dirlist.h $(FSIMG_HDR):
   @$(MAKE) -C tests TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" CROSSDEV=$(CROSSDEV)
equals to:
tests/symtab.c:
   @$(MAKE) -C tests TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" CROSSDEV=$(CROSSDEV)
tests/dirlist.h:
   @$(MAKE) -C tests TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" CROSSDEV=$(CROSSDEV)
$(FSIMG_HDR):
   @$(MAKE) -C tests TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" CROSSDEV=$(CROSSDEV)
And tests folder build multiple time concurrently. Is the right fix like this?
common:
   @$(MAKE) -C tests TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" CROSSDEV=$(CROSSDEV)

tests/symtab.c tests/dirlist.h $(FSIMG_HDR): common

Reply:
Hi, @xiaoxiang781216 

>And tests folder build multiple time concurrently. Is the right fix like this?

Hmm, I'm still not sure. Can you create another PR so that we can try?

Reply:
here: https://github.com/apache/incubator-nuttx-apps/pull/123